### ISSUE TITLE
DietPi-Software | WINE

### ISSUE THREAD

**Issue by savioruz** [MichaIng/DietPi#3914]
#### Formal software information
- Software title | WINEHQ
- Short description | Wine Is Not Emulator : to run windows application
- Official URL (if available) | https://www.winehq.org/
- GitHub URL (if available) | https://wiki.winehq.org/Source_Code
- (Official) install documentation | https://wiki.winehq.org/Debian

#### Are there similar/alternative software titles available with DietPi-Software?
No
- ...

#### What makes your requested software better than the above solutions, if available?
It is fully open source, alternative exagear (coz exagear paid) without any (feature) limitation to free users. It has a large feature range based on available apps and a large active community.
- ...

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?
According to the official install instructions there are a lot of manual steps to do for arm. DietPi could automate those steps, which saves end users a lot of time and allows installation as well to less experienced Linux users.
- ...

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
1. ... Im confused xD

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.

- Sorry, Im confused when installing wine

**Comment by Joulinar** [comment]
Hi,

many thanks for the request. According install page, WineHQ packages are for x86 architecture only; they cannot be installed on an ARM system. Means it limits the usage. It can't be used on SBC like Raspberry Pi.

**Comment by savioruz** [comment]
but twister os support wine. hmmm

**Comment by Joulinar** [comment]
well that`s how it is stated on WineHQ web site

**Comment by nonnorm** [comment]
TwisterOS supports Wine by using Box86, an emulator that translates armhf calls to x86 calls.

**Comment by savioruz** [comment]
> TwisterOS supports Wine by using Box86, an emulator that translates armhf calls to x86 calls.

if remove box86 wine not work right ?

**Comment by nonnorm** [comment]
Yes.  It does seem like you can compile wine for ARM (just aarch64 I think), but it will only run specific ported apps.

**Comment by MichaIng** [comment]
Hmm, so you need to run an x86 emulator to run a Windows emulator to run a Windows program. Probably you should simply run a program that is available for Linux or can be compiled for Linux on ARM 😄.

Jokes aside, since Windows programs are usually x86 only anyway, there is no point to get this running on ARM natively, IMO. Windows ARM64 is on the way, but it will take a looooong time until Windows-only software developers start to provide compatible software I'm afraid 😉. If there is really a need to run an x86 Windows program on e.g. Raspberry Pi 4, then qemu static emulation or even a VM should be easier, isn't it?

**Comment by savioruz** [comment]
> Hmm, so you need to run an x86 emulator to run a Windows emulator to run a Windows program. Probably you should simply run a program that is available for Linux or can be compiled for Linux on ARM 😄.
> 
> Jokes aside, since Windows programs are usually x86 only anyway, there is no point to get this running on ARM, IMO. Windows ARM64 is on the way, but it will take a looooong time until Windows-only software developers start to provide compatible software I'm afraid 😉. If there is really a need to run an x86 Windows program on e.g. Raspberry Pi 4, then qemu static emulation or even a VM should be easier, isn't it?

yep

**Comment by CactiChameleon9** [comment]
Just wanted to add, that box86 is translation (I think - check their github) instead of emulation. It tries to use native binaries making it faster than qemu.

**Comment by MichaIng** [comment]
The combination SBC + Box64 + Wine x86 is shown here: https://www.cnx-software.com/2020/11/19/box86-is-an-x86-emulator-for-raspberry-pi-and-other-32-bit-arm-platforms/
Especially for OpenGL software/games support seems to be not (yet) perfect, but the combination in general seems to work. So looks like we might want to add both, Box64 and Wine x86 and make the second depend on the first on ARM boards.

**Comment by MichaIng** [comment]
For Odroids, thanks to @meveric, there is a DEB package available for WINE via Box86: https://dietpi.com/meveric/pool/main/b/
So we can easily make it available there. For other SoCs, we need to either build and host packages or build Box86 on demand on install.

I'll just reference the Box86 request here, which IMO makes sense to implement it first: https://github.com/MichaIng/DietPi/issues/3957

**Comment by savioruz** [comment]
> For Odroids, thanks to @meveric, there is a DEB package available for WINE via Box86: https://dietpi.com/meveric/pool/main/b/
> So we can easily make it available there. For other SoCs, we need to either build and host packages or build Box86 on demand on install.
> 
> I'll just reference the Box86 request here, which IMO makes sense to implement it first: #3957

gud af

**Comment by meveric** [comment]
my package comes with a wrapper box86_wine to run box86 in combination with wine, still a command line tool, but it helps alot.

For example:
```
export MESA=1 
export WINEPREFIX=/home/odroid/.PlayOnLinux/wineprefix/Anno1602 
cd /home/odroid/.PlayOnLinux/wineprefix/Anno1602/drive_c/GOG\ Games/Anno\ 1602/
box86_wine 1602.exe
```
or as a one-liner (assuming you're already in the correct folder)
`MESA=1 WINEPREFIX=/home/odroid/.PlayOnLinux/wineprefix/Anno1602 box86_wine 1602.exe`

once again, this is standard x86/armhf software and not ODROID specific

**Comment by MichaIng** [comment]
> my package comes with a wrapper box86_wine to run box86 in combination with wine

Very nice. After upcoming DietPi release I'll play a bid with it
